### DISCUSSION TITLE
Adding annotation/label to a container-image without creating the container first

### DISCUSSION THREAD

**Discussion by tarun360** [containers/buildah#4231] [Q&A]
For adding some annotations/labels to a image, we first have to create the container from the image, and then we can add some annotations/labels using [buildah-config](https://www.mankier.com/1/buildah-config): `buildah-config [options] containerID`

But this requires first creating a container from image, then adding the required annotation/label, and then committing the changes back to an image.

But creating a container just for making some small change to the image configuration seems a unnecessary step which would take additional time and memory.

Is there a better way to add annotations/labels to a image?

**Comment by flouthoc** [comment]
Hi @tarun360 I don't think buildah can modify existing container image in storage, maybe skopeo is better candidate for this but @vrothberg and @mtrmac can confirm better if skopeo can do this or plan to do this in future or not.

**Reply by vrothberg** [reply]
Thanks for reaching out!  Skopeo does not ship such a functionality.  As pointed out, `buildah config` operates on a (build) container but a command to alter an image sounds reasonable to me.

**Reply by mtrmac** [reply]
Editing configs of images has, so far, been out of Skopeo’s scope. I agree Buildah is the appropriate tool.

---

WRT changing Buildah not to use a container in such situations, I think it’s worth exploring but as someone not very familiar with Buildah (who saw the `Containerfile` processing loops and they are _complex_), I’m worried about the extra complexity.

**Comment by rhatdan** [comment]
@nalind @flouthoc WDYT?

**Reply by nalind** [reply]
Skipping the step where we create a rootfs if we don't end up needing one would be faster in the general case, but it's kind of baked into the API.
